src/components/people.tsx

```diff
@@ -0,0 +1,177 @@
+/* eslint-disable max-len */
+import React, { useContext, useEffect, useState } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import { getPeople } from '../api';
+import { MyContext } from './state';
+import { Loader } from './Loader';
+import { Error } from './error';
+import { Person } from './person';
+import { Person as PersonType } from '../types';
+import { PeopleFilters } from './PeopleFilters';
+
+export const People = () => {
+  const { person, setPerson, error, setError, setIsLoading, isLoading } =
+    useContext(MyContext);
+  const [searchParams, setSearchParams] = useSearchParams();
+  const [originalPeople, setOriginalPeople] = useState<PersonType[]>([]);
+
+  useEffect(() => {
+    setIsLoading(true);
+    getPeople()
+      .then(fetchedPeople => {
+        setPerson(fetchedPeople);
+        setOriginalPeople(fetchedPeople);
+      })
+      .catch(() => {
+        setError('Unable to load people: ');
+      })
+      .finally(() => setIsLoading(false));
```

Comment:
This `useEffect` hook is responsible for fetching data from the API. It would be better to extract this logic into a custom hook, which would make it reusable and the component code would be cleaner.